### PR TITLE
CLN: Replace FrameOrSeriesUnion annotation by DataFrame | Series

### LINKED ISSUE #40876: STYLE, TYP frame-or-series-union check no longer relevant
This check is no longer relevant:

https://github.com/pandas-dev/pandas/blob/9ab55b4aed9018b070c939ffbdf232f99bf50f1a/.pre-commit-config.yaml#L83-L88

Since using PEP604 rewrites, this would be written as
```
x: Series | DataFrame
```
anyway.
```
x: FrameOrSeriesUnion
```
is actually just as long as the above, and arguably less explicit.

----

@simonjayhawkins thoughts on replacing all the `FrameOrSeriesUnion` occurrences, removing the alias, and removing the above check?

### PR DESCRIPTION
**Pull request by Swanand01**
- [ ] closes #40876 
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

The FrameOrSeriesUnion check was already removed. I have removed the FrameOrSeriesUnion alias, and replaced all instances of FrameOrSeriesUnion with DataFrame | Series

### CODE REVIEW

File: pandas/core/util/hashing.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
 from pandas._libs.hashing import hash_object_array
 from pandas._typing import (
     ArrayLike,
-    FrameOrSeriesUnion,
+    DataFrame | Series,
```
Comment (MarcoGorelli) [inline pandas/core/util/hashing.py:20]: DataFrame | Series doesn't need importing, just remove the FrameOrSeriesUnion import :)

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]

File: pandas/core/util/hashing.py (hunk 1 above)
Comment (Swanand01) [reply pandas/core/util/hashing.py:20]: Should I remove all other places where DataFrame | Series is being imported as well?

Comment (MarcoGorelli) [reply pandas/core/util/hashing.py:20]: yeah, `from pandas._typing import DataFrame | Series` is invalid syntax anyway

**Comment by Swanand01** [comment]
HI, I have created a new commit. Please review it. Thank you.

**Comment by MarcoGorelli** [comment]
> HI, I have created a new commit. Please review it. Thank you.

Thanks - can you fetch and merge upstream/master please?

**Comment by Swanand01** [comment]
Hi @MarcoGorelli, I seem to be stuck. I updated the forked repo on my github page by fetch and merge upstream, and tried to push changes. However, I cannot see the Compare and Pull Request button that usually appears. Initially, I tried force pushing as git would not let me push changes. Thanks.

**Comment by MarcoGorelli** [comment]
> I cannot see the Compare and Pull Request button that usually appears

If you've already opened a pull request (as you have), there's no need to open a new one, you can just add a commit and push it

**Comment by MarcoGorelli** [comment]
Looks like there's a merge conflict, could you resolve it please?

My workflow for doing this would be
```
git fetch upstream
git merge upstream/master
# resolve any conflicts
git commit -a
git push origin
```

**Pull request closed without merging**